commit 007019bb468b5991173a8a3a01714ef409f74516
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Wed May 10 08:51:09 2017 +0300

    iio: adc: ti_am335x_adc: allocating too much in probe
    
    commit 5ba5b437efaa7a502eec393c045d3bf90c92c4e9 upstream.
    
    We should be allocating enough information for a tiadc_device struct
    which is about 400 bytes but instead we allocate enough for a second
    iio_dev struct which is over 2000 bytes.
    
    Fixes: fea89e2dfcea ("iio: adc: ti_am335x_adc: use variable names for sizeof() operator")
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Jonathan Cameron <jic23@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/iio/adc/ti_am335x_adc.c b/drivers/iio/adc/ti_am335x_adc.c
index 2de1f52..62b0dec 100644
--- a/drivers/iio/adc/ti_am335x_adc.c
+++ b/drivers/iio/adc/ti_am335x_adc.c
@@ -484,7 +484,7 @@ static int tiadc_probe(struct platform_device *pdev)
 		return -EINVAL;
 	}
 
-	indio_dev = devm_iio_device_alloc(&pdev->dev, sizeof(*indio_dev));
+	indio_dev = devm_iio_device_alloc(&pdev->dev, sizeof(*adc_dev));
 	if (indio_dev == NULL) {
 		dev_err(&pdev->dev, "failed to allocate iio device\n");
 		return -ENOMEM;
